cmake/macros/macro_deal_ii_setup_target.cmake

```diff
@@ -91,7 +91,7 @@ macro(deal_ii_setup_target _target)
   endif()
 
   separate_arguments(_compile_options UNIX_COMMAND
-    "${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_${_build}}"
+    "${DEAL_II_WARNING_FLAGS} ${DEAL_II_CXX_FLAGS} ${DEAL_II_CXX_FLAGS_${_build}}"
```

Comment:
On second thought, I agree that this is what we did before but do we really want to continue to force our warning flags on downstream users?
At least, #14971 would provide the option not to do that when using the native `CMake` way.

Reply:
@masterleinad I see a couple of options.

I am currently favoring the following approach which should minimize surprises:
 - setting warning flags in the `deal_ii_initialize_cached_variables()` macro (where we set the compiler)
 - add necessary compile and link options to the `dealii::dealii` dummy target.
 - leave the `dealii::dealii_debug` and `dealii::dealii_release` targets as is.
 - reducing the `deal_ii_setup_target` macro to simply adding the `dealii::dealii` target (when merged).

Reply:
Sounds reasonable and we can do that incrementally.